commit bad098d76835c1379e1cf6afc935f8a7e050f83c
Author: Matthew Maurer <mmaurer@google.com>
Date:   Tue Oct 31 20:19:44 2023 +0000

    rust: Ignore preserve-most functions
    
    Neither bindgen nor Rust know about the preserve-most calling
    convention, and Clang describes it as unstable. Since we aren't using
    functions with this calling convention from Rust, blocklist them.
    
    These functions are only added to the build when list hardening is
    enabled, which is likely why others didn't notice this yet.
    
    Signed-off-by: Matthew Maurer <mmaurer@google.com>
    Reviewed-by: Martin Rodriguez Reboredo <yakoyoku@gmail.com>
    Reviewed-by: Alice Ryhl <aliceryhl@google.com>
    Link: https://lore.kernel.org/r/20231031201945.1412345-1-mmaurer@google.com
    [ Used Markdown for consistency with the other comments in the file. ]
    Signed-off-by: Miguel Ojeda <ojeda@kernel.org>

diff --git a/rust/bindgen_parameters b/rust/bindgen_parameters
index 552d9a85925b..a721d466bee4 100644
--- a/rust/bindgen_parameters
+++ b/rust/bindgen_parameters
@@ -20,3 +20,7 @@
 
 # `seccomp`'s comment gets understood as a doctest
 --no-doc-comments
+
+# These functions use the `__preserve_most` calling convention, which neither bindgen
+# nor Rust currently understand, and which Clang currently declares to be unstable.
+--blocklist-function __list_.*_report